### ISSUE TITLE
RNN padding support in tensorflow as MIopen already support RNN padding in rocm6.0

### ISSUE THREAD

**Issue by Yihuki** [ROCmSoftwarePlatform/tensorflow-upstream#2582]
### Issue type

Feature Request

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.17

### Custom code

No

### OS platform and distribution

Archlinux

### Python version

3.10

### Current behavior?

https://github.com/ROCm/MIOpen/pull/2343 Already adds RNN padding into MIOpen, tensorflow need to add the padded io feature.  As the RNN biLSTM or GRU still didn't allow padded input. It is easily to reproduce with keras ASR guide: https://keras.io/examples/audio/ctc_asr/ as it will report errors like
`2 root error(s) found.
  (0) INVALID_ARGUMENT:  ROCm MIOpen only supports packed input output.
         [[{{node DeepSpeech_2_1/bidirectional_1_1/forward_gru_1_1/CudnnRNNV3}}]]
         [[StatefulPartitionedCall/compile_loss/ctc_loss/ArithmeticOptimizer/ReorderCastLikeAndValuePreserving_int32_Squeeze_1/_114]]
  (1) INVALID_ARGUMENT:  ROCm MIOpen only supports packed input output.
         [[{{node DeepSpeech_2_1/bidirectional_1_1/forward_gru_1_1/CudnnRNNV3}}]]
0 successful operations.
0 derived errors ignored. [Op:__inference_one_step_on_iterator_11812]`

  https://github.com/ROCm/tensorflow-upstream/blob/f269324c13cbd9529237d2cbaee6e88c40f3a603/third_party/xla/xla/stream_executor/rocm/rocm_dnn.cc#L2897-L2909

### Standalone code to reproduce the issue

```shell
https://keras.io/examples/audio/ctc_asr/
```

**Comment by MrYoavon** [comment]
Waiting for this as well!

**Comment by philtimmes** [comment]
Any updates?

**Comment by adityas-amd** [comment]
Hi @philtimmes, https://github.com/ROCm/frameworks-internal/issues/11910 ... development is in progress, will update you as soon as possible.

**Comment by JoeNan1** [comment]
Is there any progress on this?

**Comment by adityas-amd** [comment]
this is implemented in https://github.com/ROCm/tensorflow-upstream/pull/3009

**Comment by Yihuki** [comment]
Thanks. Finally it's implemented, even though it's been a year. I'll close it now. (๑•́ ₃ •̀๑)
